File: app/src/main/java/mydebts/android/app/data/EventsSource.java
package mydebts.android.app.data;

import android.support.annotation.NonNull;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;
import mydebts.android.app.data.model.Event;

public interface EventsSource {

    Single<List<Event>> getAll();

    Single<Event> get(@NonNull Long id);

    Single<Event> insert(@NonNull Event event);

    Single<Event> update(@NonNull Event event);

    Single<Event> delete(@NonNull Event event);

    Observable<Event> inserted();

    Observable<Event> updated();

    Observable<Event> deleted();
}

File: app/src/main/java/mydebts/android/app/data/EventsSource.kt
package mydebts.android.app.data

import io.reactivex.Single
import mydebts.android.app.data.model.Event

interface EventsSource {

    fun getAll(): Single<List<Event>>

    fun get(id: Long): Single<Event>

    fun insert(event: Event): Single<Event>

    fun update(event: Event): Single<Event>

    fun delete(event: Event): Single<Event>

    fun delete(events: List<Event>): Single<List<Event>>
}
